adapters/amqp-vertx/src/test/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapterTest.java

```diff
@@ -0,0 +1,268 @@
+        // GIVEN an AMQP adapter with a configured server
+        final VertxBasedAmqpProtocolAdapter adapter = givenAnAmqpAdapter();
+        final MessageSender sender = givenAmessageSenderForAnyTenant();
+
+        // which is enabled for a tenant
+        givenAConfiguredTenant("some-tenant", true);
+
+        // IF a device sends a 'fire and forget' telemetry message
+        final ProtonDelivery delivery = mock(ProtonDelivery.class);
+        when(delivery.remotelySettled()).thenReturn(true);
+
+        final ResourceIdentifier resource = ResourceIdentifier.from("telemetry", "some-tenant", "the-device");
+        adapter.uploadMessage(new AmqpContext(delivery, getFakeMessage(), resource));
+
+        // THEN the adapter sends the message and does not wait for response from the peer.
+        verify(sender).send(any(Message.class));
+    }
+
+    /**
+     * Verifies that a request to upload telemetry message (with settled=false) results in the sender sending the
+     * message and waits for a response from the downstream peer.
+     * 
+     * AT_LEAST_ONCE delivery semantics.
+     */
+    @Test
+    public void uploadTelemetryMessageWithNotSettledDeliverySemantics() {
+        // GIVEN an adapter configured to use a user-define server.
+        final VertxBasedAmqpProtocolAdapter adapter = givenAnAmqpAdapter();
+        final MessageSender sender = givenAmessageSenderForAnyTenant();
+
+        // which is enabled for a tenant
+        givenAConfiguredTenant("enabled-tenant", true);
+
+        // IF a device send telemetry data (with un-settled delivery)
+        final ProtonDelivery delivery = mock(ProtonDelivery.class);
+        when(delivery.remotelySettled()).thenReturn(false);
+        final ResourceIdentifier resource = ResourceIdentifier.from("telemetry", "enabled-tenant", "the-device");
+        adapter.uploadMessage(new AmqpContext(delivery, getFakeMessage(), resource));
+
+        // THEN the sender sends the message and waits for the outcome from the downstream peer
+        verify(sender).sendAndWaitForOutcome(any(Message.class));
+    }
+
+    private void givenAConfiguredTenant(final String tenantId, final boolean enabled) {
+        final TenantObject tenantConfig = TenantObject.from(tenantId, enabled);
+        if (!enabled) {
+            tenantConfig.addAdapterConfiguration(new JsonObject()
+                    .put(TenantConstants.FIELD_ADAPTERS_TYPE, "amqp")
+                    .put(TenantConstants.FIELD_ENABLED, enabled));
+        }
+        when(tenantClient.get(tenantId)).thenReturn(Future.succeededFuture(tenantConfig));
+    }
+
+    private Message getFakeMessage() {
+        final Message message = mock(Message.class);
+        when(message.getContentType()).thenReturn("application/text");
+        when(message.getBody()).thenReturn(new AmqpValue("some payload"));
+        return message;
+    }
+
+    private MessageSender givenAmessageSenderForAnyTenant() {
```

Comment:
this should better be called *givenATelemetrySenderForAnyTenant* in order to point out that the sender indeed is a sender for telemetry messages.